addons/udes_security/tests/test_allowed_file_type.py

```diff
@@ -93,36 +95,45 @@ def create_allowed_file_type(self, file_type, **kwargs):
 
         return allowed_file_type
 
-    def test_assert_attachment_active_status_set_from_allowed_file_type(self):
+    def test_assert_allowed_attachment_set_active(self):
         """
-        Assert that an attachment is active if its file type is allowed, otherwise inactive.
+        Assert that an attachment with an allowed file type is set to active.
 
-        Also assert that when the active field on allowed file type is updated,
-        the active field for attachments of that file type only is also updated.
+        Also assert that the attachment is also set to active on either of the following conditions:
+            1. The relevant inactive allowed file type record is set to active
+            2. A new allowed file type record is created for the attachment's file type
         """
-        csv_active = self.atch_csv_file.active
-        rtf_active = self.atch_rtf_file.active
-        txt_active = self.atch_txt_file.active
+        self.assertTrue(self.atch_csv_file.active)
+        self.assertTrue(self.atch_rtf_file.active)
 
-        self.assertTrue(csv_active)
-        self.assertTrue(rtf_active)
-        self.assertFalse(txt_active)
+        # Set TXT allowed file type to active, unblocking TXT files
+        self.allowed_file_type_txt.active = True
+        self.assertTrue(self.atch_txt_file.active)
 
-        self.allowed_file_type_csv.active = not csv_active
-        self.assertNotEqual(self.atch_csv_file.active, csv_active)
+        # Create HTML allowed file type record, unblocking HTML files
+        self.create_allowed_file_type(("html"))
+        self.assertTrue(self.atch_html_file.active)
 
-        self.assertEqual(self.atch_rtf_file.active, rtf_active)
-        self.assertEqual(self.atch_txt_file.active, txt_active)
-
-    def test_allowed_file_type_deleted(self):
+    def test_assert_blocked_attachment_set_inactive(self):
```

Comment:
Nit: "assert" is redundant in the method name, since the purpose of test methods is to assert things. 

I tend to prefer something like:  `test_sets_blocked_attachment_inactive`, where the verb comes first, and describes what the feature under test is doing.  It is a matter of style though, so no obligation to do it that way.

Reply:
Yeah fair point, often find the naming tricky to be honest. I just got into the habit of using "assert" in the name a lot of the time. Have updated the names of the tests. :)

Reply:
I found the [GOOS book](http://www.growing-object-oriented-software.com/) really helpful for thinking about tests, if you don't mind Java.